lib/demangler-win32.js

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2018, Microsoft Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+"use strict";
+
+const utils = require('./utils'),
+    logger = require('./logger').logger,
+    DemanglerCPP = require("./demangler-cpp").Demangler;
+
+class DemanglerWin32 extends DemanglerCPP {
+    constructor(demanglerExe, compiler) {
+        super(demanglerExe, compiler);
+
+        // 0x28090 stands for:
+        //   - 0x00010 : Disable expansion of the declaration language specifier
+        //   - 0x00080 : Disable expansion of access specifiers for members
+        //   - 0x08000 : Disable enum/class/struct/union prefix
+        //   - 0x20000 : Disable expansion of __ptr64 keyword
+        this.flags = "0x28090";
+        this.allDecoratedLabels = /\?[a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*/g;
+        this.allDecoratedLabelsWithQuotes = /"\?[a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*"/;
+
+        // this is true for clang output on windows
+        // we set this to true if we see it, and false if we don't
+        // null means we haven't set it yet.
+        this.hasQuotesAroundDecoratedLabels = null;
+    }
+
+    collectLabels() {
+        this.win32RawSymbols = [];
+        for (const asmLine of this.result.asm) {
```

Comment:
More TILs here, `for (const foo of bar)` is new to me (embarrassingly, but hey! Better syntax FTW)